packages/office-ui-fabric-react/src/components/Button/Button.Props.ts

```diff
@@ -44,15 +45,23 @@ export interface IButtonProps extends React.HTMLProps<HTMLButtonElement | HTMLAn
   text?: string;
 
   /**
+  *
+  */
+  contextualMenuItems?: IContextualMenuItem[];
+
+  /**
    * The button icon shown in command or hero type.
-   * Deprecated at v1.2.3, to be removed at >= v2.0.0. Use IconButton instead.
    */
   icon?: string;
 
   /**
+   * The button icon shown to the right of the text
+   */
+  splitIcon?: string;
```

Comment:
This probably should be an enum and we should consider some options.

```
menuType: {
default,
none,
chevron,
split
}
```

I don't know if you need split. If a button has an onClick or href, and a menu, it has to be a split button.